ssr/next-js.md

```diff
@@ -0,0 +1,385 @@
+    *   [Server Components](#server-components)
+*   [Optimizations](#optimizations)
+    *   [Images](#images)
+    *   [Dynamic import](#dynamic-import)
+
+## Installation
+
+There are two ways to set up the Next.js project.
+
+### Automatic Installation:
+
+Starting from **Next.js version 12,** Next.js provides an automatic installation feature that allows you to set up a new project with just one command. Here's how you can use the automatic installation feature:
+
+1\. Install the **create-next-app** package globally: Run the following command to install the **create-next-app** package globally on your system:
+
+```plaintext
+npm install -g create-next-app
+```
+
+2\. Create a new Next.js project: Navigate to the directory where you want to create your new Next.js project, and run the following command:
+
+```plaintext
+npx create-next-app
+```
+
+3\. Select a template (optional): After running the above command, you'll be prompted to choose a template for your project. You can select one of the available templates or press Enter to skip this step and use the default template.
+
+```plaintext
+What is your project named? my-app
+Would you like to use TypeScript with this project? No / Yes
+Would you like to use ESLint with this project? No / Yes
+Would you like to use Tailwind CSS with this project? No / Yes
+Would you like to use `src/` directory with this project? No / Yes
+Use App Router (recommended)? No / Yes
+Would you like to customize the default import alias? No / Yes
+```
+
+4\. Navigate into the project directory: Once the project is created, navigate into the project directory by running the following command:
+
+```plaintext
+cd my-app
+```
+
+5\. Start the development server: Run the following command to start the Next.js development server:
+
+```plaintext
+npm run dev 
+```
+
+### Manual Installation:
+
+1\. Create a new project folder: Create a new directory for your Next.js project. You can do this through your operating system's command line or terminal.  
+2\. Initialize a new project: Navigate to the project folder using the command line or terminal and run the following command to initialize a new Node.js project:
+
+```plaintext
+npm init -y
+```
+
+This command creates a new **package.json** file for your project with default settings.
+
+3\. Install Next.js. Run the following command to install Next.js and React:
+
+```plaintext
+npm install next react react-dom
+```
+
+4\. Open `package.json` and add the following `scripts`:
+
+```plaintext
+{
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  }
+}
+```
+
+5\. Create Next.js pages: In Next.js, pages are created as individual files inside a folder called **pages**. Create a new **pages** folder in your project directory and create your first page. For example, you can create a file named **index.js** inside the **pages** folder with the following content:
+
+```plaintext
```

Comment:
Please use `javascript` here instead of the `plaintext`. And review other parts of this document